### ISSUE TITLE
magnitude limits for galaxy knots models in Run 2.0i

### ISSUE THREAD

**Issue by rmandelb** [LSSTDESC/DC2-production#264]
As discussed in #261 and #263 , we have run into resource issues with the large input catalogs for Run 2.0i.  This problem has arisen now because cosmoDC2 (which provides the Run 2.0 input catalog) fills in more of the faint galaxy population and the galaxies at 1<z<3 which were missing in protoDC2 (which provided the Run 1.2 input catalog).

Today a related issue has arisen: the knots model, which is used to produce more complex morphology in Run 2.0i, further inflates the instance catalog sizes (since it turns galaxy disk components into a disk component and a knot component).  This inflation has essentially eaten up all of the gains due to optimizations that were made in order to enable us to use the full-depth input catalogs.

Since there were several scientific motivations for the knots model in Run 2.0i -- including more complex morphology for shear estimation and inclusion of morphologies that violate some deblender assumptions -- we would like a solution to the resource problem that preserves that science.  The current plan is to *avoid simulating knots on very faint galaxies*.  If something is going to be detected at very low significance, or not detected at all, we are proposing to go back to the bulge+disk model rather than including knots.  To be more specific, 88% of the knots are in galaxies fainter than i=27; 95% are in galaxies fainter than i=26. Reverting to bulge+disk for those very faint galaxies will have very little impact on the science, while essentially fixing these resource issues due to the instance catalog sizes.

From a science perspective, this means we still get to test 

a) issues related to blending of brighter galaxies with faint galaxies, and 
b) issues related to shear estimation and blending for galaxies that are well-detected and have more complex morphologies.

The only issue we are missing is complex morphology for the very faint galaxies that may be blended with brighter ones.  This compromise still lets us meet some of the key science goals of DC2 Run 2.0i, while enabling interesting comparisons between Run 2.0p and 2.0i (since the former will have no knots, not even on the brighter galaxies).

Francois @EiffL has confirmed it should be possible to make the necessary changes to instance catalogs within our current time-frame, and is hard at work on them.  Comments on i=26 or 27 for the cut are welcome.  Using i=26 should cover anything that makes it into the WL, LSS, or CL sample analyses.

**Comment by wmwv** [comment]
Making a cut of a minimum brightness for using a knot model sounds reasonable to me.

I don't have deeply informed thoughts about 26 vs. 27 mag.

**Comment by rmjarvis** [comment]
Agreed.  I think we only care about the knots for the actual galaxies that we are trying to measure shapes on.  Not the ones that we want to be messing up the nearby background.  So I'm fine to not do knots on galaxies fainter than 26.5.  (I think that's probably already conservative for what will make it into the lensing sample.)

**Comment by danielsf** [comment]
I just tested Francois' code.  Cutting at i-band magnitude of 27 reduces the number of entries in the knots InstanceCatalog by a factor of 10.  That should be ample.

**Comment by egawiser** [comment]
If the disks for faint (i>26) galaxies are realistically small, this seems unlikely to cause a major lack of realism.  (I would worry about large disks because then it would cause a huge drop in surface brightness, essentially removing faint objects from being detectable, even if they would still contribute to photometric errors due to blending with undetected objects.)  It sounds like there isn't a lot of run-time gain for eliminating knots at i>26 though, so I support choosing i>27 to be on the safer side.

**Comment by rmandelb** [comment]
OK.  Thanks, all; I'm hearing no objections whatsoever to 27, with 26.5 probably being OK as well.  Since 27 seems to make a huge difference already, let's just go with that.  Will leave this open for objections to this proposal until later tonight.

@fjaviersanchez thanks for the plot!   It's useful to see something quantitative that is consistent with our intuition/experience.

**Comment by rmandelb** [comment]
@jchiang87 @danielsf - is there anything else needed in practice to implement this i=27 knot cut?  Since there were no objections to the proposal, I think this discussion is done, so we can close this issue unless it needs to stay open as a reminder to do something.

**Comment by jchiang87** [comment]
Sorry, we should have reported that the PR with the changes had been merged:  https://github.com/LSSTDESC/sims_GCRCatSimInterface/pull/62
